NLog.Web.AspNetCore/LayoutRenderers/AspNetItemValueLayoutRenderer.cs

```diff
@@ -79,18 +80,21 @@ protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
             {
                 return;
             }
-            var context = HttpContextAccessor.HttpContext;
 
-            if (context == null)
+            var context = HttpContextAccessor.HttpContext;
+#if !ASP_NET_CORE
+            Func<System.Collections.IDictionary, string, object> getVal = (items, key) => 
             {
-                return;
-            }
-
-            Func<string, object> getVal = k => context.Items[k];
-
-            var value = PropertyReader.GetValue(Variable, getVal, EvaluateAsNestedProperties);
+                return items?.Count > 0 && items.Contains(key) ? items[key] : null;
+            };
+#else
+            Func<IDictionary<object, object>, string, object> getVal = (items, key) =>
```

Comment:
why Funcs and not private methodes?

Reply:
No special reason. Think I was just keeping the existing coding-style.

Reply:
I really prefer private method then. Let clutter.

Reply:
By private methods do you mean normal methods or local methods (c# 7.0) ?

Reply:
normal please.

Reply:
Done